osinfo/src/config.rs

```diff
@@ -64,14 +77,22 @@ impl OsInfo {
             os_info::Bitness::X64 => Bitness::Bit64,
             _ => Bitness::Unknown,
         };
+        let version = os_info.version().to_string();
+        let name = Some(
+            match &architecture {
+                Some(arch) => format!("{family} {version} {arch}"),
+                None => format!("{family:?} {version}"),
+            }
+        );
```

Comment:
I think, if I understand correctly, that we only want this surfaced and returned from the `export` operation, not `get`:

- When directly invoking with `dsc resource get`, this property doesn't mean anything / is just noise.
- When invoking for the get operation with `dsc config get`, the name of the resource instance is already defined by the `name` field of the resource instance definition.
- When invoking for the export operation, we _do_ want to surface `_name`, so it can be hoisted to the effective `name` of the exported instance definition.